openfeature/api.py

```diff
@@ -30,6 +31,7 @@
 
 _evaluation_context = EvaluationContext()
 
+_hooks_lock = threading.RLock()
```

Comment:
If we know we are not going to be reentrant, it's faster to use a `threading.Lock`

Reply:
yeah, let's use a normal `Lock` I don't think it is going to be reentrant. If it changes, we can easily change it to `RLock` 😄